api/views/pois.py

```diff
@@ -0,0 +1,104 @@
+from sqlalchemy.sql.expression import func
+from flask import Blueprint, request, jsonify
+import json
+from datetime import date
+from dateutil.parser import parse
+
+mod = Blueprint('POIS', __name__)
+
+def poi_links_media_stories(poi_dict):
+    """
+    Gets additional links, media, and stories for a POI
+    """
+    poi_links = Link.query.filter(Link.poi_id == poi_dict['_id'])
+    poi_media = Media.query.filter(Media.poi_id == poi_dict['_id'])
+    poi_stories = Story.query.join(StoryPOI, Story.id == StoryPOI.story_id).filter(StoryPOI.poi_id == poi_dict['_id'])
+    poi_dict['links'] = [j.to_dict() for j in poi_links]
+    poi_dict['media'] = [k.to_dict() for k in poi_media]
+    poi_dict['stories'] = [l.to_dict() for l in poi_stories]
+    return poi_dict
+
+@app.route('/pois', methods=['GET'])
+def pois_get_by_map_year_or_story():
+    map_year, story_id = request.args.get('map_year'), request.args.get('story_id')
+    if map_year is not None and story_id is None:
+        # Find POIs by map_year
+        pois = POI.query.filter(POI.map_year == int(map_year))
+        if pois.count() == 0:
+            return create_response(status=404, message='No POIs found')
+    elif story_id is not None and map_year is None:
+        # Find POIs by story_id
+        pois = POI.query.join(StoryPOI, POI.id == StoryPOI.poi_id).filter(StoryPOI.story_id == int(story_id))
+        if pois.count() == 0:
+            return create_response(status=404, message='No POIs found')
+    # Add associated links, media, and stories to each POI, then create a list of all POIs
+    pois_list = [poi_links_media_stories(i.to_dict()) for i in pois]
+    return create_response({'pois': pois_list})
+
+@app.route('/pois/<poi_id>', methods=['GET'])
+def pois_get_by_id(poi_id):
+    poi_by_id = POI.query.get(int(poi_id))
+    if poi_by_id is None:
+        return create_response(status=404, message='No POI found')
+    # Find associated links, media, and stories
+    poi_result = poi_links_media_stories(poi_by_id.to_dict())
+    return create_response({'poi': poi_result})
+
+@app.route('/pois', methods=['POST'])
+def pois_post():
+    data = request.get_json()
+    # Check that all required parameters exist
+    if not all(i is not None for i in [data['name'], data['description'], data['map_year'], data['x_coord'], data['y_coord']]):
+        return create_response(status=422, messsage='POI not created; missing one or more required input parameters')
+    for j in data['links']:
+        if j['link_url'] is None:  # Check for link url in each link
+            return create_response(status=422, message='POI not created; missing link URL')
+    for k in data['media']:
+        if k['content_url'] is None:  # Check for content url in each medium
+            return create_response(status=422, message='POI not created; missing media URL')
+    for l in data['story_ids']:
+        if Story.query.get(l) is None:  # Check if story exists for each story_id

```

Comment:
this comment is okay, but move it above the `if` statement. it might be more clear that a certain story might not exist if you say something along the lines of "Make sure story exists for each story_id"